ethereum/executionlayer/src/test/java/tech/pegasys/teku/ethereum/executionlayer/ExecutionLayerManagerImplTest.java

```diff
@@ -234,9 +235,46 @@ public void builderGetHeaderGetPayload_shouldReturnHeaderAndPayloadViaEngineWhen
     final UInt64 slot = executionPayloadContext.getForkChoiceState().getHeadBlockSlot();
     final BeaconState state = dataStructureUtil.randomBeaconState(slot);
 
-    prepareBuilderGetHeaderResponse(executionPayloadContext, false);
+    final UInt256 builderValue =
+        prepareBuilderGetHeaderResponse(executionPayloadContext, false).getValue();
+    final ExecutionPayload localExecutionPayload =
+        prepareEngineGetPayloadResponse(executionPayloadContext, builderValue, slot);
+
+    // we expect result from the local engine
+    final ExecutionPayloadHeader expectedHeader =
+        spec.getGenesisSpec()
+            .getSchemaDefinitions()
+            .toVersionBellatrix()
+            .orElseThrow()
+            .getExecutionPayloadHeaderSchema()
+            .createFromExecutionPayload(localExecutionPayload);
+
+    // we expect local engine header as result
+    final HeaderWithFallbackData expectedResult =
+        HeaderWithFallbackData.create(
+            expectedHeader,
+            new FallbackData(localExecutionPayload, FallbackReason.LOCAL_BLOCK_VALUE_WIN));
+    assertThat(executionLayerManager.builderGetHeader(executionPayloadContext, state))
+        .isCompletedWithValue(expectedResult);
+    verifyFallbackToLocalEL(slot, executionPayloadContext, expectedResult);
+  }
+
+  @Test
+  public void builderGetHeaderGetPayload_shouldReturnEnginePayloadWhenValueLowerButChallengeWon() {
+    // Setup requires local payload to have at lest 50% value of builder's to win
+    executionLayerManager = createExecutionLayerChannelImpl(true, false, 50);
+    setBuilderOnline();
+
+    final ExecutionPayloadContext executionPayloadContext =
+        dataStructureUtil.randomPayloadExecutionContext(false, true);
+    final UInt64 slot = executionPayloadContext.getForkChoiceState().getHeadBlockSlot();
+    final BeaconState state = dataStructureUtil.randomBeaconState(slot);
+
+    final UInt256 builderValue =
+        prepareBuilderGetHeaderResponse(executionPayloadContext, false).getValue();
     final ExecutionPayload localExecutionPayload =
-        prepareEngineGetPayloadResponse(executionPayloadContext, UInt256.MAX_VALUE, slot);
+        prepareEngineGetPayloadResponse(
+            executionPayloadContext, builderValue.multiply(51).divide(100), slot);
```

Comment:
50 would be the boundary right? could just multiply by boundary here...

Reply:
It's in percents

Reply:
I wanted to have 2 tests cases side by side one where value is a bit higher and another where it's a bit smaller. And I converted the first to default value, but it's not passed from the same place in tests, so it's not correct say it's default, which I don't like

Reply:
ok but the boundary is 50%, so it'd be nice to also test the boundary and make sure >= doesnt turn into >

Reply:
ahhh got you
the reason I didn't make it because it could change with even/odd number
Of course, I could use not random to be sure

Reply:
and we test >= on old test

Reply:
ok fair :)